src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java

```diff
@@ -181,6 +181,30 @@ void toManyArgs(int aArg1, int aArg2, int aArg3, int aArg4, int aArg5,
                     int aArg6, int aArg7, int aArg8, int aArg9)
     {
     }
+
+    /**
+     * Very long url with https: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java
+     */
+    public void testHttps() {}
+
+    /**
+     * Very long url with http: http://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java
+     */
+    public void testHttp() {}
+
+    /**
+     * Very long url with ftp: ftp://ftp.dlink.ru/pub/D-Link_Solutions/D-Link_Solutions_for_Business.pdf
+     */
+    final double testFtp = 99.99;
+
+
+    // Very long url with href: href://ftp.dlink.ru/pub/D-Link_Solutions/D-Link_Solutions_for_Business.pdf
```

Comment:
Please remove this case, it is not a real case.

Reply:
We can keep it and it should be violation as it is unknown protocol "href".
Our config supports only well known

Reply:
Why is it not a violation?

Reply:
We support href URLs, as you can see here:

https://github.com/checkstyle/checkstyle/blob/173925e9a52370dd80e66b2e847e1b0ff1b1a7eb/src/main/resources/google_checks.xml#L45-L49

it is used for `<a href="someurl"></a>`

Reply:
`Very long url with hre` --> `placing 'href' word  in any long line suppress violation `

Reply:
I see, then we will need to remove `href` from the ignorePattern and also update the `a href` to `<a href="`.

Reply:
i have updated the regex, now it will be only valid when it is a html hyper link:

https://github.com/checkstyle/checkstyle/blob/a41e7bffd380c18838ea34ae1513684f8821f8db/src/main/resources/google_checks.xml#L45-L49

Reply:
`|href` was for cases when line is starting by attribute `href="...."`, tag a is on previous line

Reply:
Nobody complained about this suppression for 10 years, let's not change it.

Reply:
done:
https://github.com/checkstyle/checkstyle/blob/8d93b36bc13a0161531563036cca811fc0a4840c/src/main/resources/google_checks.xml#L45-L50

Reply:
after re-adding the `|href`, this line is not considered as violation:

https://github.com/checkstyle/checkstyle/blob/8d93b36bc13a0161531563036cca811fc0a4840c/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java#L201

as you said here: https://github.com/checkstyle/checkstyle/pull/14943#discussion_r1630637409. This line should cause violation but because `|href` is present, it is bypassing the rule.

Failing Test case result:

```
Error message at position 2 of 'expected' does not match actual message
expected: C:\OpenSource\checkstyle\src\it\resources\com\google\checkstyle\test\chapter4formatting\rule44columnlimit\InputLineLength.java:201: Line is longer than 100 characters (found 120).
but was : Audit done.
```

Reply:
So should we keep `|href` or not?

Reply:
ok, remove `|href|`.
BUT lets do this in separatate issue, to let users see that we changed it and it was in releasenotes, not a secret side effect under JSNI.

in this PR lets focus on JSNI only and no other updates to config or mapping table.

Reply:
separate issue: https://github.com/checkstyle/checkstyle/issues/14955